cdk/backend.ts

```diff
@@ -163,7 +164,25 @@ const coapDockerfilePath = path.join(
 	'containers',
 	'coap',
 )
-const coapHash = await hashFolder(coapDockerfilePath)
+const coapSimulatorDownloadUrl = (
+	await getCoAPHealthCheckSettings({
+		ssm,
+		stackName: STACK_NAME,
+	})
+).simulatorDownloadURL.toString()
+const { headers, ok, status } = await fetch(coapSimulatorDownloadUrl, {
+	method: 'HEAD',
+})
+if (!ok) throw new Error(`Failed to download CoAP simulator: ${status}`)
```

Comment:
status (a number) is not very helpful to understand the error. Print the message.

```suggestion
if (!ok) throw new Error(`Failed to download CoAP simulator: ${await res.text()}`)
```